bench/micro/Bench/Database/LSMTree/Internal/Merge.hs

```diff
@@ -0,0 +1,391 @@
+        , mergeMappend = Just (onDeserialisedValues ((+) @Word64))
+        }
+      -- different key and value sizes
+    , benchMerge configWord64
+        { name         = "insert-large-keys-x4"  -- potentially long keys
+        , nentries     = map (`div` 10) $ scaled $ replicate 4 1
+        , finserts     = 1
+        , randomKey    = first serialiseKey . randomByteStringR (6, 4000)
+        }
+    , benchMerge configWord64
+        { name         = "insert-mixed-vals-x4"  -- potentially long values
+        , nentries     = map (`div` 10) $ scaled $ replicate 4 1
+        , finserts     = 1
+        , randomValue  = first serialiseValue . randomByteStringR (0, 4000)
+        }
+    , benchMerge configWord64
+        { name         = "insert-page-x4"  -- 1 page
+        , nentries     = map (`div` 10) $ scaled $ replicate 4 1
+        , finserts     = 1
+        , randomValue  = first serialiseValue . randomByteStringR (4056, 4056)
+        }
+    , benchMerge configWord64
+        { name         = "insert-page-plus-byte-x4"  -- 1 page + 1 byte
+        , nentries     = map (`div` 10) $ scaled $ replicate 4 1
+        , finserts     = 1
+        , randomValue  = first serialiseValue . randomByteStringR (4057, 4057)
+        }
+    , benchMerge configWord64
+        { name         = "insert-huge-vals-x4"  -- 1-5 pages
+        , nentries     = map (`div` 10) $ scaled $ replicate 4 1
+        , finserts     = 1
+        , randomValue  = first serialiseValue . randomByteStringR (10_000, 20_000)
+        }
+      -- common UTxO scenarios
+    , benchMerge configUTxO
+        { name         = "utxo-x4"  -- like tiering merge
+        , nentries     = scaled $ replicate 4 1
+        , finserts     = 1
+        , fdeletes     = 1
+        }
+    , benchMerge configUTxO
+        { name         = "utxo-x4-uneven"
+        , nentries     = scaled [1, 3, 1.5, 2.5]
+        , finserts     = 1
+        , fdeletes     = 1
+        }
+    , benchMerge configUTxO
+        { name         = "utxo-x4-lastlevel"
+        , nentries     = scaled $ replicate 4 1
+        , finserts     = 1
+        , fdeletes     = 1
+        , mergeLevel   = Merge.LastLevel
+        }
+    , benchMerge configUTxO
+        { name         = "utxo-x4+1-min-skewed-lastlevel"  -- live levelling merge
+        , nentries     = scaled [1, 1, 1, 1, 4]
+        , finserts     = 1
+        , fdeletes     = 1
+        , mergeLevel   = Merge.LastLevel
+        }
+    , benchMerge configUTxO
+        { name         = "utxo-x4+1-max-skewed-lastlevel"  -- live levelling merge
+        , nentries     = scaled [1, 1, 1, 1, 16]
+        , finserts     = 1
+        , fdeletes     = 1
+        , mergeLevel   = Merge.LastLevel
+        }
+    ]
+  where
+    totalentries = 50_000 :: Word64
+
+    -- For each run, a significant part of all possible keys are in use, e.g.
+    -- half of them for 4 runs of equal size. This leads to many collisions.
```

Comment:
Could you elaborate? What does having 4 runs of equal size have to do with it? And by "in use" do you mean ones that are actually in the run?

Reply:
Yes, that's what I mean. And the number of runs is relevant because they affect the number of entries per run (the 50k entries get distributed across the runs), and therefore the ratio of keys that are present to possible keys.

I will rework the description, especially since I also added an argument to the function as you suggested.